pkg/config/validater.go

```diff
@@ -0,0 +1,220 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"fmt"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/mitchellh/mapstructure"
+	"gopkg.in/yaml.v2"
+
+	"istio.io/mixer/pkg/aspect"
+	"istio.io/mixer/pkg/expr"
+)
+
+type (
+	// ValidaterFinder is used to find specific underlying validaters
+	// Manager registry and adapter registry should implement this interface
+	// so ConfigValidaters can be uniformly accessed
+	ValidaterFinder interface {
+		Find(name string) (aspect.ConfigValidater, bool)
+	}
+)
+
+// NewValidater returns a validater given component validaters
+func NewValidater(managerFinder ValidaterFinder, adapterFinder ValidaterFinder, strict bool, exprValidater expr.Validater) *Validater {
+	return &Validater{
+		managerFinder: managerFinder,
+		adapterFinder: adapterFinder,
+		strict:        strict,
+		exprValidater: exprValidater,
+		validated:     &Validated{},
+	}
+}
+
+type (
+	// Validater is the Configuration validator
+	Validater struct {
+		managerFinder ValidaterFinder
+		adapterFinder ValidaterFinder
+		strict        bool
+		exprValidater expr.Validater
+		validated     *Validated
+	}
+
+	// Validated store validated configuration
+	// It has been validated as internally consistent and correct
+	Validated struct {
```

Comment:
would just `Valid` work here?  or is there something else we could call this that reflects its purpose (and we could document that it is only created by config that validates)? Maybe something like `Mixer` ?